### PR TITLE
Add support for INF and NAN

### LINKED ISSUE #33: Support for INF/NAN in rbx_xml
BillboardGui's MaxDistance is INF and this presumably causes issues. I assume this happens for nan too.

### PR DESCRIPTION
**Pull request by Kampfkarren**
Closes #33

### CODE REVIEW

File: rbx_xml/src/types/numbers.rs [hunk 1]
```diff
@@ -8,7 +8,56 @@ use crate::{
...
+                    writer.write_characters("NAN")?;
+                } else {
+                    writer.write_characters(value)?;
+                }
+                writer.write(XmlWriteEvent::end_element())?;
+
+                Ok(())
+            }
+
+            #[allow(clippy::zero_divided_by_zero)]
```
Comment (Kampfkarren) [inline rbx_xml/src/types/numbers.rs]: Neither std::$rust_type::NAN nor std::<$rust_type>::NAN compile, so 0.0 / 0.0 is needed.

Comment (LPGhatguy) [reply rbx_xml/src/types/numbers.rs]: I'd expect `std::<$rust_type>::NAN` to compile, interesting.

Comment (LPGhatguy) [reply rbx_xml/src/types/numbers.rs]: at least there's a bug for it: https://github.com/rust-lang/rust/issues/48067

Comment (Kampfkarren) [reply rbx_xml/src/types/numbers.rs]: I think this isn't bad enough for readability to matter?

Comment (LPGhatguy) [reply rbx_xml/src/types/numbers.rs]: Yeah, it's proabbly fine.

File: rbx_xml/src/types/numbers.rs [hunk 2]
```diff
@@ -90,4 +139,22 @@ mod test {
...
+        test_util::test_xml_serialize::<Float32Type, _>(
+            r#"<float name="foo">-INF</float>"#,
+            &std::f32::NEG_INFINITY,
+        );
+
+        test_util::test_xml_serialize::<Float32Type, _>(
+            r#"<float name="foo">NAN</float>"#,
+            &std::f32::NAN,
+        );
+    }
```
Comment (LPGhatguy) [inline rbx_xml/src/types/numbers.rs]: Can you add deserialization tests for these too?

Comment (Kampfkarren) [reply rbx_xml/src/types/numbers.rs]: I'll work on it tomorrow.

Comment (Kampfkarren) [reply rbx_xml/src/types/numbers.rs]: Done.

**Pull request merged**